### DISCUSSION TITLE
jupyter-book CLI failure on notebook execution error?

### DISCUSSION THREAD

**Discussion by brian-rose** [executablebooks/meta#892] [Q&A]
Hi all,

At [Project Pythia](https://projectpythia.org) we are working on integrating JupyterBook with a CI system for regular health-checking of notebook-based content.

For this purpose, it would be really useful to have `jupyter-book build` report a failure if an execution error is encountered in a notebook. Is there a way to enforce this?

I'm aware of the `--warningiserror` option [in the docs here](https://jupyterbook.org/en/stable/reference/cli.html#jupyter-book-build), and that accomplishes the goal of failing on notebook execution error. But the problem is that it will also fail for other reasons unrelated to code in the notebooks.

Is there a more fine-grained way of setting the fail condition for `jupyter-book build`?

**Comment by chrisjsewell** [comment]
Heya, just n the latest myst-nb there is the config `nb_execution_raise_on_error`: https://myst-nb.readthedocs.io/en/latest/configuration.html#configuration

**Comment by brian-rose** [comment]
@chrisjsewell fantastic, that seems to be just what I need!

So to pass this through via `jupyter-book build` to enforce this rule for all notebooks, would I put this in the `_config.yml` file something like this?
```
sphinx:
  config:
    nb_execution_raise_on_error: True
```
